Managers/PlayerManager.gd

```diff
@@ -54,7 +82,12 @@ func set_player(in_player: Player) -> void:
 
 func create_persistent_data() -> void:
 	player_persistent_data = PlayerPersistentData.new()
-	
+
+func init_data() -> void:
+	player = null
+	is_player_initial_position_set = false
+	player_position = Vector2i(-1,-1)
+	player_room = null
```

Comment:
Need to put `is_map_movement_allowed = true` here to reinitialize the flag and allow movement whenever the game is recreated

Reply:
this solves the issue for me too